Cython/Compiler/ModuleNode.py

```diff
@@ -2261,8 +2261,17 @@ def generate_import_star(self, env, code):
     def generate_module_init_func(self, imported_modules, env, code):
         code.enter_cfunc_scope(self.scope)
         code.putln("")
-        header2 = "PyMODINIT_FUNC init%s(void)" % env.module_name
-        header3 = "PyMODINIT_FUNC %s(void)" % self.mod_init_func_cname('PyInit', env)
+        code.putln("#ifdef CYTHON_NO_PYINIT_EXPORT")
+        code.putln("/* define this to PyObject * manually because PyMODINIT_FUNC")
+        code.putln("* adds __declspec(dllexport) to it's definition.")
+        code.putln("*/")
+        code.putln("#define __Pyx_PyMODINIT_FUNC PyObject *")
+        code.putln("#else")
+        code.putln("#define __Pyx_PyMODINIT_FUNC PyMODINIT_FUNC")
+        code.putln("#endif")
+        code.putln("")
+        header2 = "__Pyx_PyMODINIT_FUNC init%s(void)" % env.module_name
```

Comment:
The module init function actually "returns" `void` in Py2 and also comes with a special case for C++ mode. See
https://github.com/python/cpython/blob/b97001ef46e599642affde61e31abace71486703/Include/pyport.h#L744-L813

Reply:
oh, I see now going to change it now.